### PR TITLE
OpenAPI 3

### PR DESCRIPTION
**Pull request by eguzki**
OpenAPI 3 support PHASE 1
* `3scale import openapi` command now supports openapi 3
* The command will perform the very same tasks, no matter openapi is 2.0 or 3.0.X.
* README updated
* https://github.com/Nexmo/oas_parser parser used
* Currently, not all features of OAS3 are supported. Limitations:
    * Only first `server.url` element in `servers` list parsed as private url. As OpenAPI `basePath` property, `server.url` element's path component will be used.
    * Toolbox will not parse servers in path item or operation objects.
    * Supported security schemes: apiKey, oauth2 (any flow type).
    * Multiple flows in security scheme object not supported.

### CODE REVIEW

**Comment by miguelsorianod** [comment]
Out of curiosity, have you evaluated this alternative? https://github.com/kevindew/openapi3_parser

**Comment by eguzki** [comment]
> Out of curiosity, have you evaluated this alternative? https://github.com/kevindew/openapi3_parser

No, I haven't. I found https://github.com/Nexmo/oas_parser from https://openapi.tools/#parsers and it has a simple, yet useful interface. It lacks validation, but it is just a simple json schema validation, two lines implementation.

File: lib/3scale_toolbox/openapi/oas3.rb [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+      def operations
+        @operations ||= parse_operations
+      end
+
+      def security
+        @security ||= parse_security
+      end
+
+      def service_backend_version
```
Comment (miguelsorianod) [inline lib/3scale_toolbox/openapi/oas3.rb]: Are we coupling 3scale specifics to the oas3/swagger classes? If that's intended then :+1: and will keep adding 3scale specific modifications/methods there

Comment (eguzki) [reply lib/3scale_toolbox/openapi/oas3.rb]: Initially we had one class `ThreescaleOpenAPISpec` designed for that abstraction you mention. It did not serve very well. 

I decided to remove it and have oas3/swagger classes implement the same interface.  

```ruby
##                                                                                                                             
#                                                                                                                                   
# OpenAPI object                                                                                                                       
# * OpenAPI.title -> string                                                                                                            
# * OpenAPI.description -> string                                                                                                      
# * OpenAPI.version -> string                                                                                                          
# * OpenAPI.base_path -> string                                                                                                        
# * OpenAPI.host -> string                                                                                                             
# * OpenAPI.scheme -> string                                                                                                           
# * OpenAPI.operation -> array of operation hash                                                                                       
#   * operation hash properties                                                                                                     
#     * :verb                                                                                                                       
#     * :path                                                                                                                       
#     * :description                                                                                                                
#     * :operation_id                                                                                                               
# * OpenAPI.security ->  security hash                                                                                                 
#   * security hash properties                                                                                                      
#     * :id -> string                                                                                                               
#     * :type -> string                                                                                                             
#     * :name -> string                                                                                                             
#     * :in_f -> string                                                                                                             
#     * :flow -> symbol (:implicit_flow_enabled, :direct_access_grants_enabled, :service_accounts_enabled, :standard_flow_enabled)  
#     * :scopes -> array of string                                                                                                  
# * OpenAPI.service_backend_version -> string ('1','2','oidc')                                                                         
# * OpenAPI.set_server_url -> def(spec, url)                                                                                           
# * OpenAPI.set_oauth2_urls-> def(spec, scheme_id, authorization_url, token_url)                                                       
```

So my idea was to think about `ThreeScaleToolbox::OpenAPI::OAS3` and `ThreeScaleToolbox::OpenAPI::Swagger` not as OpenAPI parsers, but as 3scale openapi interface implementations.

File: .dependency_decisions.yml [hunk 2]
```diff
@@ -69,6 +69,12 @@
 - - :approve
   - bundler
   - :who: 
-    :why: MIT License https://github.com/bundler/bundler/blob/master/LICENSE.md 
+    :why: MIT License https://github.com/bundler/bundler/blob/master/LICENSE.md
     :versions: []
     :when: 2019-05-28 09:24:54.882581954 Z
+- - :whitelist
+  - CC0 1.0 Universal
+  - :who: Jeff Kaufmann and Richard Fontana (Red Hat Legal)
```
Comment (eguzki) [inline .dependency_decisions.yml]: Red Hat legal just confirmed CCO licenses are acceptable

**Pull request merged**